### PR TITLE
Upload SBOMs in the promotion pipeline

### PR DESCRIPTION
**Pull request by chmeliik**
Builds on https://github.com/redhat-appstudio/tssc-dev-multi-ci/pull/23

Instructions for testing are in the README update

Haven't tested in Jenkins yet

### CODE REVIEW

**Comment by chmeliik** [comment]
> Haven't tested in Jenkins yet

Have tested it now, appears to work! :tada:

**Comment by chmeliik** [comment]
Fixed formatting

File: generated/gitops-template/jenkins/Jenkinsfile [hunk 1]
```diff
@@ -12,6 +12,13 @@ pipeline {
         COSIGN_SECRET_KEY = 'dummy'
         /* Used to verify the image signature and attestation */
         COSIGN_PUBLIC_KEY = credentials('COSIGN_PUBLIC_KEY')
+        /* URL of the BOMbastic api host (e.g. https://sbom.trustification.dev) */
+        TRUSTIFICATION_BOMBASTIC_API_URL = credentials('TRUSTIFICATION_BOMBASTIC_API_URL')
+        /* URL of the OIDC token issuer (e.g. https://sso.trustification.dev/realms/chicken) */
+        TRUSTIFICATION_OIDC_ISSUER_URL = credentials('TRUSTIFICATION_OIDC_ISSUER_URL')
+        TRUSTIFICATION_OIDC_CLIENT_ID = credentials('TRUSTIFICATION_OIDC_CLIENT_ID')
+        TRUSTIFICATION_OIDC_CLIENT_SECRET = credentials('TRUSTIFICATION_OIDC_CLIENT_SECRET')
+        TRUSTIFICATION_SUPPORTED_CYCLONEDX_VERSION = credentials('TRUSTIFICATION_SUPPORTED_CYCLONEDX_VERSION')
```
Comment (chmeliik) [inline generated/gitops-template/jenkins/Jenkinsfile:21]: Only one of these is an actual secret (TRUSTIFICATION_OIDC_CLIENT_SECRET), but I don't know if there's any better way to set this configuration :thinking:

Comment (jduimovich) [reply generated/gitops-template/jenkins/Jenkinsfile:21]: > Only one of these is an actual secret (TRUSTIFICATION_OIDC_CLIENT_SECRET), but I don't know if there's any better way to set this configuration 🤔

The rhtap/env.sh holds the per-repo environment variables which are loaded by the pipeline scripts 
These env vars were originally defined as the values for the pipeline parameters and was the portable way to allow user defined env vars. Each CI has its own mechanism so we can consider evolving pipelines to use those.

Comment (chmeliik) [reply generated/gitops-template/jenkins/Jenkinsfile:21]: The thing is that all these values are properties of a specific RHTAP deployment, they're not something that a user should have to set. IMO they should ideally be set by the platform engineer

Comment (chmeliik) [reply generated/gitops-template/jenkins/Jenkinsfile:21]: In that sense, credentials might be the right choice in Jenkins (the unfortunate part is that defining them as credentials means Jenkins will censor them in the logs, so the URLs are hidden)

Comment (jduimovich) [reply generated/gitops-template/jenkins/Jenkinsfile:21]: > The thing is that all these values are properties of a specific RHTAP deployment, they're not something that a user should have to set. IMO they should ideally be set by the platform engineer

I agree. Yes, the site install defaults are set by the platform engineer when they fork the templates repo so in theory they fork the templates and update the defaults, Currently the can set the URLs for site things like quay url, glab url, org default and could do the same for env. Its not "clean" and I would prefer the templates don't need to be customized (so they are easier to update) and the values get installed into RHDH as some sort of site profile which could have multiples (per team etc) and reduce the duplication per repo. For not as the secrets across CIs can be "org" or global based and shared in multiple repos this is good enough

**Comment by chmeliik** [comment]
Rebased on main. I noticed that adding the promote steps to `templates/data.yaml` only updated the Jenkins promote pipeline, not the other CIs. I'll look into that, but this PR should be ready for review anyway

**Comment by chmeliik** [comment]
Rebased on main

**Comment by chmeliik** [comment]
Also updated the `hack/ghub-set-{org,}-vars` and `hack/glab-set-vars` scripts with the TRUSTIFICATION_ vars

**Comment by jduimovich** [comment]
@chmeliik  
Tried this on github actions without REKOR or TUFs so no transparency log (which is supported via IGNORE_REKOR and setting the REKOR and TUFs 
```
export REKOR_HOST=
export TUF_MIRROR=
```
Testing was done via checkout of this pr  `git pull upstream/28/head`  and then` running ci-test.sh `  so should be repeatable.
You may be able to see [](https://github.com/jduimovich/tssc-dev-gitops/actions/runs/11822582856) but the error is below
Will retry with a REKOR_HOST and TUF_MIRROR when my install comes up
```
Step: download-sbom-from-url-in-attestation
Results: /__w/tssc-dev-gitops/tssc-dev-gitops/results/download-sbom-from-url-in-attestation
Getting attestation for ***/bootstrap:github-be414ecb8ddf3ef706b5f4cad0[42](https://github.com/jduimovich/tssc-dev-gitops/actions/runs/11822582856/job/32939936770#step:7:43)8194cdf634ce
Failed to verify any attestation type. Errors:
  
  Command: cosign verify-attestation --type=slsaprovenance02 --key /tmp/download-sbom-workdir.xQGJOs/cosign.pub ***/bootstrap:github-be414ecb8ddf3ef706b5f4cad0428194cdf634ce
  
  Error: no matching attestations: signature not found in transparency log
  main.go:74: error during command execution: no matching attestations: signature not found in transparency log
```
  
  Command: cosign verify-attestation --type=slsaprovenance1 --key /tmp/download-sbom-workdir.xQGJOs/cosign.pub ***/bootstrap:github-be414ecb8ddf3ef706b5f4cad0428194cdf634ce
  
  Error: no matching attestations: signature not found in transparency log
  main.go:74: error during command execution: no matching attestations: signature not found in transparency log
Error: Process completed with exit code 1.

**Comment by chmeliik** [comment]
@jduimovich are you setting IGNORE_REKOR somewhere other than here? It defaults to false https://github.com/jduimovich/tssc-dev-gitops/blob/dccd3599ed97ec5cba6cc779f3a5287be328dddc/rhtap/env.sh#L44

**Comment by jduimovich** [comment]
It works with a proper REKOR_HOST and TUFS_MIRROR
It also works with the FAIL_IF_TRUSTIFICATION_NOT_CONFIGURED set to false so QE will not be blocked

**Pull request merged**